src/Dependencies/Collections/ImmutableSegmentedList.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    internal static class ImmutableSegmentedList
+    {
+        /// <inheritdoc cref="ImmutableList.Create{T}()"/>
+        public static ImmutableSegmentedList<T> Create<T>()
+            => ImmutableSegmentedList<T>.Empty;
+
+        /// <inheritdoc cref="ImmutableList.Create{T}(T)"/>
+        public static ImmutableSegmentedList<T> Create<T>(T item)
+            => ImmutableSegmentedList<T>.Empty.Add(item);
+
+        /// <inheritdoc cref="ImmutableList.Create{T}(T[])"/>
+        public static ImmutableSegmentedList<T> Create<T>(params T[] items)
+            => ImmutableSegmentedList<T>.Empty.AddRange(items);
+
+        /// <inheritdoc cref="ImmutableList.CreateBuilder{T}()"/>
+        public static ImmutableSegmentedList<T>.Builder CreateBuilder<T>()
+            => ImmutableSegmentedList<T>.Empty.ToBuilder();
+
+        /// <inheritdoc cref="ImmutableList.CreateRange{T}(IEnumerable{T})"/>
+        public static ImmutableSegmentedList<T> CreateRange<T>(IEnumerable<T> items)
+            => ImmutableSegmentedList<T>.Empty.AddRange(items);
+
+        /// <inheritdoc cref="ImmutableList.ToImmutableList{TSource}(IEnumerable{TSource})"/>
+        public static ImmutableSegmentedList<T> ToImmutableSegmentedList<T>(this IEnumerable<T> source)
+        {
+            if (source is ImmutableSegmentedList<T> existingList)
+                return existingList;
+
+            return ImmutableSegmentedList<T>.Empty.AddRange(source);
+        }
+
+        /// <inheritdoc cref="ImmutableList.ToImmutableList{TSource}(ImmutableList{TSource}.Builder)"/>
+        public static ImmutableSegmentedList<T> ToImmutableSegmentedList<T>(this ImmutableSegmentedList<T>.Builder builder)
```

Comment:
this one seems odd.  if i have a builder on it, i'll have to equivalent methods named very similarly.

Reply:
I agree, but it helps with drop-in replacements.
https://github.com/dotnet/runtime/blob/4c92aef2b08f9c4374c520e7e664a44f1ad8ce56/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList.cs#L68-L78